app/controllers/hq/sponsorships_controller.rb

```diff
@@ -0,0 +1,37 @@
+class Hq::SponsorshipsController < HqController
+
+  def inactivate
+    sponsorship = Sponsorship.find(params[:id])
+
+    @sponsorship_inactivator = InactivateSponsorship.new(sponsorship: sponsorship,
+                                                         end_date: params[:sponsorship][:end_date])
+    inactivate_sponsorship and redirect_to hq_sponsor_path(sponsorship.sponsor_id)
+  end
+
+  def destroy
+    sponsorship = Sponsorship.find(params[:id])
+
+    @sponsorship_destructor = DestroySponsorship.new(sponsorship)
+
+    destroy_sponsorship and redirect_to hq_sponsor_path(sponsorship.sponsor_id)
+  end
```

Comment:
Yeah I second that @gmunumel @razorcd .  I had long wondered why the `SponsorshipsController` was not using the `sponsorship_id`.  I think @nik mentioned it before as well.  This was long overdue